File: 5. Longest Palindromic Substring/CodeVersion2.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if len(s) == 1:
            return s
        
        maxStr = s[0]

        def expand(l, r):
            while l >= 0 and r <= len(s) - 1 and s[l] == s[r]:
                l -= 1
                r += 1
            return s[l+1:r]

        for i in range(len(s)-1):
            odd = expand(i,i)
            even = expand(i, i+1)

            if len(odd) > len(maxStr):
                maxStr = odd
            if len(even) > len(maxStr):
                maxStr = even
            
        return maxStr

File: 5. Longest Palindromic Substring/CodeVersion2.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    if (s.length === 1) {
        return s;
    }
    
    let maxStr = s[0];

    function expand(l, r) {
        while (l >= 0 && r <= s.length - 1 && s[l] === s[r]) {
            l--;
            r++;
        }
        return s.substring(l + 1, r);
    }

    for (let i = 0; i < s.length - 1; i++) {
        let odd = expand(i, i);
        let even = expand(i, i + 1);

        if (odd.length > maxStr.length) {
            maxStr = odd;
        }
        if (even.length > maxStr.length) {
            maxStr = even;
        }
    }
    return maxStr;
};
